### PR TITLE
bpo-39357: Remove buffering parameter of bz2.BZ2File

### PR DESCRIPTION
**Pull request by vstinner**
Remove the buffering parameter of bz2.BZ2File. Since Python 3.0, it
was ignored and using it was emitting a DeprecationWarning. Pass an
open file object to control how the file is opened.

https://bugs.python.org/issue39357

### CODE REVIEW

File: Lib/bz2.py
Comment (serhiy-storchaka) [inline Lib/bz2.py]: When remove a parameter, make all following parameters keyword-only. This will help to catch errors like `BZ2File(filename, 'w', 0)`. Please add also a test for such example.

File: Doc/library/bz2.rst
Comment (serhiy-storchaka) [inline Doc/library/bz2.rst]: Should not the whole directive be removed?

Comment (vstinner) [reply Doc/library/bz2.rst]: done

File: Lib/bz2.py
Comment (vstinner) [reply Lib/bz2.py]: Good idea: done

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by vstinner** [comment]
@serhiy-storchaka: I updated my PR, would you mind to review it again? I converted compresslevel parameter to a keyword-only.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by vstinner** [comment]
Oh, I didn't notice that you already approved it :-) I rebased and squashed commits after your approved my PR, but it's the same code. I also did that to update the commit message ;-) So I understand that you approved the PR. Thanks! I will merge it once the CI tests pass.

**Comment by vstinner** [comment]
@serhiy-storchaka: Thanks again for the review, I didn't spot that the positional argument order changed! It's a nice idea to convert compresslevel to a keyword-only parameter.

**Pull request merged**
